simpeg_drivers/utils/tile_estimate.py

```diff
@@ -0,0 +1,186 @@
+from simpeg_drivers.utils.utils import (
+    active_from_xyz,
+    simpeg_group_to_driver,
+    tile_locations,
+)
+
+
+class TileEstimator:
+    def __init__(self, input_file: str | Path | InputFile):
+        if isinstance(input_file, str | Path):
+            input_file = InputFile.read_ui_json(input_file)
+
+        if not isinstance(input_file, InputFile):
+            raise ValueError(
+                "Input must be a path to a UI JSON file or an InputFile instance."
+            )
+
+        self.params = TileParameters.build(input_file)
+
+    def start(self) -> SimPEGGroup:
+        """
+        Run the tile estimator.
+        """
+        driver = simpeg_group_to_driver(self.params.simulation, self.params.geoh5)
+        mesh = driver.inversion_mesh.mesh
+        locations = driver.inversion_data.locations
+        active_cells = active_from_xyz(
+            driver.inversion_mesh.entity, locations, method="nearest"
+        )
+        results = {}
+        counts = fibonacci_series(15)[2:]
+        for count in tqdm(counts, desc="Estimating tiles:"):
+            tiles = tile_locations(
+                locations,
+                count,
+                method="kmeans",
+            )
+            # Get the median tile
+            ind = np.argsort([len(tile) for tile in tiles])[int(count / 2)]
+            survey, _, _ = driver.inversion_data.create_survey(
+                mesh=mesh, local_index=tiles[ind]
+            )
+            driver.params.tile_spatial = int(count)
+            local_sim, local_map = driver.inversion_data.simulation(
+                mesh,
+                active_cells,
+                survey,
+                padding_cells=driver.params.padding_cells,
+                tile_id=count,
+            )
+            results[count] = float(survey.nD) * local_map.shape[0] * count * 8 * 1e-9
+
+        print(f"Computed tile sizes: {results}")
+
+        optimal = self.estimate_optimal_tile(results)
+
+        new_out_group = self.params.simulation.copy(copy_children=False)
+        driver.params.tile_spatial = optimal
+        driver.params.out_group = new_out_group
+        new_out_group.options = driver.params.to_dict(ui_json_format=True)
+        new_out_group.metadata = None
+
+        if self.params.out_group is not None:
+            new_out_group.parent = self.params.out_group
+
+        if self.params.render_plot:
+            figure = self.plot(results, locations, optimal)
+            path = self.params.geoh5.h5file.parent / "tile_estimator.png"
+            figure.savefig(path)
+            new_out_group.add_file(path)
+            figure.savefig(path)
+
+        if self.params.monitoring_directory is not None:
+            monitored_directory_copy(self.params.monitoring_directory, new_out_group)
+
+        return new_out_group
+
+    @staticmethod
+    def estimate_optimal_tile(results: dict) -> int:
+        """
+        Estimate the optimal number of tiles to use using a circle fit.
```

Comment:
Help me out here!  Describe what is being fit to a circle and why it's optimal.

Reply:
Point of max curvature in curve marks the point where we don't gain much in reducing the problem size, but start increasing the amount of work for many more tiles. So of an estimate for the point of diminishing return.